File: web/cb/mortgage.php
<?
  $arg = explode(" ", $args);
  $P = $arg[0];
  $r = $arg[1];
  $n = $arg[2];
  $d = $arg[3];

  if(!$P || !$r || !$n)
    die("Usage: !mortgage <principle> <rate%> <number of years> [downpayment].  Example: !mortgage 230000 5.5 30");

  $P -= $d;
  $r = round($r / 1200, 5);
  $n = $n * 12;

  $a = $P * (pow((1 + $r),$n) * $r) / (pow((1 + $r), $n) - 1);

  print("Monthly Payment: $".round($a, 2));
?>

File: hooks/mortgage.rb
m = match_args(/(\d+(?:\.\d+)?)\s+(\d+(?:\.\d+)?)%?\s+(\d+)(?:\s+(\d+(?:\.\d+)?))?/,
      '<principle> <rate%> <number of years> [downpayment]   Example: !mortgage 230000 5.5 30')
_,p,r,n,d = m.to_a.map(&:to_f)

p -= d
r = (r / 1200.0)
r = r.round(5)
n = n * 12.0

a = ((p * (1 + r) ** n) * r) / ((1 + r) ** n - 1)

reply "Monthly payment: $#{a.round(2)}"
